onnxruntime/core/providers/webgpu/webgpu_context.cc

```diff
@@ -279,7 +279,11 @@ Status WebGpuContext::Run(ComputeContext& context, const ProgramBase& program) {
   uint32_t x = program.DispatchGroupSizeX();
   uint32_t y = program.DispatchGroupSizeY();
   uint32_t z = program.DispatchGroupSizeZ();
-  ORT_RETURN_IF_ERROR(program_mgr_->NormalizeDispatchGroupSize(x, y, z));
+
+  // Skip normalization for indirect dispatch since dimensions are determined by the indirect buffer
```

Comment:
could check if `SetIndirectDispatchTensor` and `SetDispatchGroupSize` are both called.

if (validation_mode >= basic && dispatch_tensor != nullptr && (dispatch_x != 0 || dispatch_y != 0 || dispatch_z != 0) {
  ERROR
}

Reply:
Done